src/types.md

```diff
@@ -0,0 +1,183 @@
+# Questions about your types
+
+## My 'class' needs references to lots of other things in order to do its job, and that conflicts with other things which also hold similar references. What do I do?
```

Comment:
This question leaves out the crucial aspect of mutability. How about this?

"My 'class' needs mutable references to other things to do its job. Other classes need mutable references to these things too. What do I do?"

(Also, here, I'm consistently using "class" for the entities that contain the references, and "things" for the the entities that the references refer to -- I think that maybe helps reinforce that distinction.)